src/main/src/test/java/org/geoserver/data/test/SystemTestData.java

```diff
@@ -346,28 +346,53 @@ XStreamPersister createXStreamPersister() {
 
     /**
      * Adds a workspace to the test setup.
-     * 
+     *
      * @param name The name of the workspace.
      * @param uri The namespace uri associated with the workspace.
      */
     public void addWorkspace(String name, String uri, Catalog catalog) {
-        
+        addWorkspace(name, uri, false, catalog);
+    }
+
+    /**
+     * Adds a workspace to the test setup. If the workspace and namespace already exists
+     * they will be updated, otherwise they will be created.
+     *
+     * @param name name of the workspace
+     * @param uri the namespace URI associated with the workspace
+     * @param isolated TRUE if the workspace and associated namespace are isolated
+     * @param catalog the catalog were to store \ update the workspace and associated namespace
+     */
+    public void addWorkspace(String name, String uri, boolean isolated, Catalog catalog) {
+
+        // let's see if the workspace already exists
         WorkspaceInfo ws = catalog.getWorkspaceByName(name);
         if (ws == null) {
+            // new workspace, we need to create a new one
             ws = catalog.getFactory().createWorkspace();
             ws.setName(name);
+            ws.setIsolated(isolated);
             catalog.add(ws);
+        } else {
```

Comment:
Given the method name, I would have preferred to see an exception (trying to add an already existing workspace should result in an error?), but the old code was confusing too, was ignoring the add and was eventually aligning the namespace.
So I guess it's all right :)